packages/grafana-eslint-rules/README.md

```diff
@@ -111,3 +111,34 @@ const getStyles = (theme: GrafanaTheme2) => ({
 ### `theme-token-usage`
 
 Used to find all instances of `theme` tokens being used in the codebase and emit the counts as metrics. Should **not** be used as an actual lint rule!
+
+### `no-untranslated-literals`
+
+Check if strings are marked for translation.
+
+Right now, we only check if a string is wrapped up by the `Trans` tag. We currently do not apply this rule to props, attributes or similars.
+
+```tsx
+// No untranslated strings ❌
+<InlineToast placement="top" referenceElement={buttonRef.current}>
+  Copied
+</InlineToast>
+
+// Wrap strings with `<Trans />` ✅
+<InlineToast placement="top" referenceElement={buttonRef.current}>
+  <Trans i18nKey="clipboard-button.inline-toast.success">Copied</Trans>
+</InlineToast>
+
+// Mark for translation when using variables ✅
+const SearchTitle = ({ term }) => (
+  <Trans i18nKey="search-page.results-title">
+    Results for <em>{{ term }}</em>
+  </Trans>
+);
+//Or
+const serviceName = service.name;
+<Trans i18nKey="login.services.sing-in-with-prefix">Sign in with {{ serviceName }}</Trans>
+
+```
```

Comment:
this could be it's own subheading, e.g.
#### Passing variables to translations
```tsx
// Bad ❌
const SearchTitle = ({ term }) => (
  <div>
    Results for <em>{term}</em>
  </div>
);
// Good ✅
const SearchTitle = ({ term }) => (
  <div>
    <Trans i18nKey="search-page.results-title">
      Results for <em>{{ term }}</em>
    </Trans>
  <div>
);
// Good ✅
// ...same thing but `t()` example
```

Reply:
Ok. Instead of adding this like `<Trans/>`, I expanded the explanation of what we are checking and mentioned that we also aim for the props and attributes to be translated with `t()`, and then added examples of how to do that. Would this be ok?

> Right now, we only check if a string is wrapped up by the `Trans` tag. We currently do not apply this rule to props, attributes or similar, but we also ask for them to be translated with the `t()` function.